### ISSUE TITLE
Hidden Units

### ISSUE THREAD

**Issue by scJazz2** [MegaMek/megamek#38]
If, Player A moves a unit into a "Point Blank" shot situation with a Hidden Unit from Player B.

Player B decides to make the attack.

Player B ends up not firing anything and clicks done with no weapons fire.

Player A ends up in their move phase again with no way to click done or continue or move on.

The game is stuck. No way to continue

No errors in logs.

[PBShot_Cancelled.sav.gz](https://github.com/MegaMek/megamek/files/143492/PBShot_Cancelled.sav.gz)

**Comment by scJazz2** [comment]
Actually... probably has something to do with 2 Hidden Units right next to each other.

In earlier Comment the Foot Infantry did not take a shot. The Longtom said OK and then did not fire anything.

[autosave.sav.gz](https://github.com/MegaMek/megamek/files/143515/autosave.sav.gz)

**Comment by arlith** [comment]
I dunno, I can't reproduce this.  If I load the first (PBShot_Cancelled.sav.gz) saved game, it's in the movement phase as normal and I can make a move.  If I load the second game (autosave.sav.gz) and move the Spider into 0704 and say "No" to the PBS for Infantry and then "Yes" for the Long Tom, then just click done, I don't have any issues.  The game continues on like it should.

I also tried this in repeated configurations with the Spider, moving it back and force along the hidden units and I never once ran into a case where I got stuck on movement for the Spider.  Did you try clicking new hexes to move to?

I did however find a few bugs, namely hidden units who take PBS's weren't marked as such, and hence could make multiple PBS's.  I also made a few other visual enhancements (units facing is correct now, and the board view does a better job of highlighting the attacker and target).
